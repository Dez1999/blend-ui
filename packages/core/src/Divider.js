import React from "react";
import styled, { css } from "styled-components";
import { space } from "styled-system";
import { default as styledProps } from "@styled-system/prop-types";
import PropTypes from "prop-types";

const normalDivider = css`
  border-bottom-style: ${props => props.borderStyle || "solid"};
  border-bottom-width: ${props => props.borderWidth || "1px"};
  border-color: ${props => props.borderColor || props.theme.colors.baseMuted};
`;
const textInDivider = css`
  display: flex;
  align-items: center;
  text-align: center;
  ::before,
  ::after {
    content: "";
    flex: 1;
    border-bottom-style: ${props => props.borderStyle || "solid"};
    border-bottom-width: ${props => props.borderWidth || "1px"};
    border-color: ${props => props.borderColor || props.theme.colors.baseMuted};
  }
  ::before {
    margin-right: 1.25em;
  }
  ::after {
    margin-left: 1.25em;
  }
`;

/*
const DividerElement = styled("div")`
  ${space}
  border: 0;
  border-bottom-style: ${props => props.borderStyle || "solid"};
  border-bottom-width: ${props => props.borderWidth || "1px"};
  border-color: ${props => props.borderColor || props.theme.colors.baseMuted};
`;
*/
const DividerElement = styled("div")`
  ${space}
  border: 0;
  ${props => (props.children ? textInDivider : normalDivider)}
`;
const Divider = ({ as, height, color, style, ...props }) => {
  //console.log("DIV ", props);
  const opts = {
    borderWidth: height || null,
    borderStyle: style || null,
    borderColor: color || null,
    ...props,
  };
  let dividerType = as;
  if (props.children) {
    dividerType = "div";
  }
  return <DividerElement as={dividerType} {...opts} />;
};

Divider.displayName = "Divider";

Divider.defaultProps = {
  as: "hr",
};

Divider.propTypes = {
  ...styledProps.color,
  ...styledProps.space,
  ...styledProps.layout,
  ...styledProps.border,
  as: PropTypes.oneOf(["hr", "div"]),
};

export default Divider;
