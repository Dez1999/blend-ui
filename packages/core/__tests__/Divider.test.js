import React from "react";
import renderer from "react-test-renderer";
import Divider from "../src/Divider";

test("Divider renders correctly", () => {
  const tree = renderer.create(<Divider />).toJSON();
  expect(tree).toMatchSnapshot();
});
