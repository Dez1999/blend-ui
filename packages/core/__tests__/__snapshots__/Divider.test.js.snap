// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider renders correctly 1`] = `
<hr
  className="Divider__DividerElement-qfr82c-0 dJhULX"
/>
`;
